From 8cc03924a36c572c2908e659e624f44636dc2b33 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Imre=20Vad=C3=A1sz?= <imre@vdsz.com>
Date: Sun, 24 Jul 2016 21:11:29 +0200
Subject: [PATCH 067/100] if_iwm - Fix inverted logic in iwm_tx().

The PROT_REQUIRE flag in should be set for data frames above a certain
length, but we were setting it for !data frames above a certain length,
which makes no sense at all.

Taken-From: OpenBSD, Linux iwlwifi
---
 sys/dev/netif/iwm/if_iwm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/dev/netif/iwm/if_iwm.c b/sys/dev/netif/iwm/if_iwm.c
index 95e158a..b5817c4 100644
--- a/sys/dev/netif/iwm/if_iwm.c
+++ b/sys/dev/netif/iwm/if_iwm.c
@@ -3423,7 +3423,7 @@ iwm_tx(struct iwm_softc *sc, struct mbuf *m, struct ieee80211_node *ni, int ac)
 		flags |= IWM_TX_CMD_FLG_ACK;
 	}
 
-	if (type != IEEE80211_FC0_TYPE_DATA
+	if (type == IEEE80211_FC0_TYPE_DATA
 	    && (totlen + IEEE80211_CRC_LEN > vap->iv_rtsthreshold)
 	    && !IEEE80211_IS_MULTICAST(wh->i_addr1)) {
 		flags |= IWM_TX_CMD_FLG_PROT_REQUIRE;
-- 
2.7.2

